packages/internal/src/lib-chainStorage.js

```diff
@@ -152,3 +152,19 @@ export async function makeStorageNodeChild(storageNodeRef, childName) {
   return E(storageNode).makeChildNode(childName);
 }
 harden(makeStorageNodeChild);
+
+/**
+ *
+ * @param {import('@endo/far').ERef<StorageNode>} storageNode
+ * @param {import('@endo/far').ERef<Marshaller>} marshaller
+ * @returns {(value: unknown) => Promise<void>}
+ */
+export const makeMarshallToStorage = (storageNode, marshaller) => {
+  return value =>
+    E(marshaller)
+      .serialize(value)
+      .then(serialized => {
+        const encoded = JSON.stringify(serialized);
+        return E(storageNode).setValue(encoded);
+      });
+};
```

Comment:
Ugh, these [names](https://github.com/endojs/endo/issues/1248)...
```suggestion
export const makeSerializeToStorage = (storageNode, marshaller) => {
  return async value => {
    const marshalled = await E(marshaller).serialize(value);
    const serialized = JSON.stringify(marshalled);
    return E(storageNode).setValue(serialized);
  };
};
```

But also, what is the benefit of defining such a function in lib-chainStorage? I'd be happiest if this package were imported only from the vat that calls `makeChainStorageRoot` (i.e., the one with access to the bridge device) to which other vats just sent simple messages.

Also, if sending an object to a remote Marshaller to get input for JSON.stringify serialization on the way to chain storage is a common pattern, then I'd also like to save round trips by incorporating the JSON.stringify serialization into a logical wrapper of either Marshaller `serialize` or [less preferably] StorageNode `setValue` and getting the initiator out of the path between them, e.g. `storageNode.setValue(E(marshaller).marshalAndSerialize(value))`[^1][^2] or `storageNode.serializeAndSetValue(E(marshaller).serialize(value))`).

[^1]: Extending StorageNode `setValue`'s input signature from `(value: string)` to `(value: ERef<string>)`.
[^2]: Where `marshalAndSerialize` takes a passable, encodes it (to CapData/smallcaps/etc.), serializes the result (to JSON text), and returns the result—and could/would be more tersely named `serialize` if that were not currently claimed by just the first [encoding] step.

Reply:
+1 on the names. I can't invest in until after Vault release.

The code change LGTM and I'll probably incorporate it into another PR. Not worth retriggering deployment-test rn. 

The round trips is a problem. I think @dckc 's meant to help https://github.com/Agoric/agoric-sdk/pull/6646